oauth-common/src/test/java/io/strimzi/kafka/oauth/jsonpath/CustomCheckTest.java

```diff
@@ -0,0 +1,222 @@
+        "@.roles.client-roles.kafka noneof ['kafka-admin', 'kafka-user']", "false",
+        "@.sub anyof ['username']", "false",
+        "@.missing anyof [null, 'username']", "false",
+        "@.missing noneof [null, 'username']", "false",
+        "@.aud noneof [null, 'username']", "true",
+        "@.aud noneof ['kafka', 'something']", "false",
+        "@.aud anyof ['kafka', 'something']", "true",
+        "!(@.aud anyof ['kafka', 'something'])", "false",
+        "!(@.aud noneof [null, 'username'])", "false",
+        "((@.custom == 'some-custom-value' || 'kafka' in @.aud) && @.exp > 1000)", "false",
+        "((('kafka' in @.aud || @.custom == 'custom-value') && @.exp > 1000))", "false",
+        "@.exp =~ /^6[0-9][0-9]$/", "true",
+        "@.custom =~ /custom-.+/", "true",
+        "@.custom =~ /ustom-.+/", "false",
+        "@.custom =~ /CUSTOM-.+/i", "true",
+        "@.iss =~ /https:\\/\\/auth-server\\/.+/", "true",
+        "@.iss =~ /https:\\/\\/auth-server\\//", "false",
+        "@.custom == 'custom-value' && @.exp > 1000", "false",
+        "@.custom == 'custom-value' || @.exp > 1000", "true",
+        "(@.custom == 'custom-value' || @.custom == 'custom-value2')", "true",
+        "@.missing == null", "false",
+        "@.missing != null", "true",
+        "@.missing", "false",
+        "!@.missing", "true",
+        "@.custom", "true",
+        "@.sub && @.custom == 'custom-value'", "true",
+        "@.custom == 'custom-value' && @.roles.client-roles.kafka", "true",
+        "@.['custom'] == 'custom-value'", "true",
+        "@.custom in ['custom-value','custom-value2','custom-value3']", "true",
+        "@.custom-level && @.custom-level nin [1,2,3]", "true",
+        "@.missing nin [1,2,3]", "true",
+        "@.missing && @.missing nin [1,2,3]", "false",
+        "\"kafka-user\" in @.['roles'].['client-roles'].['kafka']", "true",
+        "@.roles.client-roles.kafka && \"kafka-admin\" nin @.roles.client-roles.kafka", "true"
+    };
+
+    /*
+     * These are queries supported by Jayway implementation but not supported by our implementation.
+     * It's mostly the lack of whitespace.
+     */
+    final String[] poorWhitespaceQueries = {
+        "@.custom-level!=-1", "true",
+        "@.iat==null", "false",
+        "@.custom=~/CUSTOM-.+/i", "true",
+        "@.custom=='custom-value' && @.exp>1000", "false",
+        "@.custom=='custom-value'&&@.exp>1000", "false"
+    };
+
+    /*
+     * These queries return a different result in Jayway JsonPath implementation
+     *
+     * In the case of 'anyof' and 'noneof' Jayway seems to silently fail the evaluation when the referenced
+     * attribute is missing, which results in EXPRESSION and !(EXPRESSION) to both effectively evaluate to false.
+     */
+    final String[] buggyJaywayQueries = {
+        "!(@.missing anyof [null, 'username'])", "true",
+        "!(@.missing noneof [null, 'username'])", "true"
+    };
```

Comment:
Are there issues opened against the jayway impl for these?

Reply:
Not yet, on my TODOs ...

Reply:
https://github.com/json-path/JsonPath/issues/662